plugins/package-managers/conan/src/main/kotlin/Conan.kt

```diff
@@ -182,64 +177,37 @@ class Conan(
             }
 
             val jsonFile = createOrtTempDir().resolve("info.json")
-            if (config.lockfileName != null) {
-                verifyLockfileBelongsToProject(workingDir, config.lockfileName)
-                ConanCommand.run(
-                    workingDir,
-                    "info", definitionFile.name,
-                    "-l", config.lockfileName,
-                    "--json", jsonFile.absolutePath
-                ).requireSuccess()
-            } else {
-                ConanCommand.run(
-                    workingDir,
-                    "info", definitionFile.name,
-                    "--json", jsonFile.absolutePath,
-                    *DUMMY_COMPILER_SETTINGS
-                ).requireSuccess()
-            }
-
-            val pkgInfos = parsePackageInfos(jsonFile)
-            jsonFile.parentFile.safeDeleteRecursively()
-
-            val packageList = removeProjectPackage(pkgInfos, definitionFile.name)
-            val packages = parsePackages(packageList, workingDir)
-            val projectInfo = findProjectPackageInfo(pkgInfos, definitionFile.name)
-
-            val dependenciesScope = Scope(
-                name = SCOPE_NAME_DEPENDENCIES,
-                dependencies = parseDependencyTree(pkgInfos, projectInfo.requires, workingDir)
-            )
-            val devDependenciesScope = Scope(
-                name = SCOPE_NAME_DEV_DEPENDENCIES,
-                dependencies = parseDependencyTree(pkgInfos, projectInfo.buildRequires, workingDir)
-            )
 
-            val projectPackage = generateProjectPackage(projectInfo, definitionFile, workingDir)
+            val handlerResults = handler.process(jsonFile, definitionFile, workingDir, config.lockfileName)
```

Comment:
The `jsonFile` does not seem to be properly deleted now anymore. But also see my comment below about a proposed API change to avoid that problem altogether.